### PR TITLE
Added new interface for checking empty stuff

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
Added new interface to enable classes to be checked for beeing empty.

### CODE REVIEW

File: Editor/EditorReflectionUtils.cs [hunk 1]
```diff
@@ -79,6 +79,36 @@ public static bool ClassExists(string assemblyName, string className)
             return assembly.GetType(className) != null;
         }
 
+        public static IEnumerable<MemberInfo> GetAllFieldsAndProperties(object value)
```
Comment (Gusinuhe) [inline Editor/EditorReflectionUtils.cs]: Missing docs

File: Runtime/Utils/Audio/IAudioData.cs [hunk 2]
```diff
@@ -6,7 +7,7 @@ namespace Innoactive.Creator.Core.Audio
     /// This class provides audio data in form of an AudioClip. Which also might not be loaded at the time needed,
     /// check first if there can be one provided.
     /// </summary>
-    public interface IAudioData
+    public interface IAudioData : ICanBeEmpty
```
Comment (Gusinuhe) [inline Runtime/Utils/Audio/IAudioData.cs:10]: It's sad 😟

**Review by Gusinuhe — APPROVED** [review]

File: Editor/EditorReflectionUtils.cs (hunk 1 above)
Comment (SimonTheSourcerer) [reply Editor/EditorReflectionUtils.cs]: Should not be in there, removed it

File: Runtime/Utils/Audio/IAudioData.cs (hunk 2 above)
Comment (SimonTheSourcerer) [reply Runtime/Utils/Audio/IAudioData.cs:10]: :(

Comment (SimonTheSourcerer) [reply Runtime/Utils/Audio/IAudioData.cs:10]: https://media1.giphy.com/media/M28rUlcjueKUE/giphy.gif?cid=ecf05e47ntozs4clmprw4elsbodblyfety0eely22yhkwu14&rid=giphy.gif

**Review by tomwim — APPROVED** [review]

**Pull request merged**